### PR TITLE
Fix issue self-hosted databases with ec2 hostnames fail to create

### PR DESCRIPTION
**Pull request by greedy52**
Fix an issue that types.NewDatabaseV3 throws the error `database xxxx" AWS account ID is empty` when the URI ends with `amazonaws.com` like an ec2 host.

### CODE REVIEW

**Comment by greedy52** [comment]
@marcoandredinis It seems this issue would only happen through the new WebAPI. The URI for self-hosted databases should have a port at the end. So maybe the WebAPI should do some similar checks.
https://github.com/gravitational/teleport/blob/b85de1a7ee960444f9292a6730caa1b4635cd82f/lib/service/cfg.go#L994-L999

**Comment by greedy52** [comment]
Sorry for reopening this. I think I will make a separate PR for pre-checks for the webapi.
https://github.com/gravitational/teleport/blob/3fd1cca5b2bcc1c91da03eb18f44ab3a26c82492/lib/service/cfg.go#L948-L1032

File: api/utils/aws/endpoint.go [hunk 1]
```diff
@@ -59,6 +59,12 @@ func IsMemoryDBEndpoint(uri string) bool {
 	return isAWSServiceEndpoint(uri, MemoryDBSServiceName)
 }
 
+// IsKeyspacesEndpoint returns true if input URI is an AWS Keyspaces endpoint.
+// https://docs.aws.amazon.com/keyspaces/latest/devguide/programmatic.endpoints.html
+func IsKeyspacesEndpoint(uri string) bool {
+	return strings.HasPrefix(uri, "cassandra.") && IsAWSEndpoint(uri)
```
Comment (marcoandredinis) [inline api/utils/aws/endpoint.go:65]: There's also `cassandra-fips` (fedramp/govcloud only?)
Should we relax the condition to `"cassandra"` or add an an `OR` to `"cassandra-"` ?

File: api/utils/aws/endpoint.go [hunk 2]
```diff
@@ -576,8 +582,7 @@ const (
 // CassandraEndpointURLForRegion returns a Cassandra endpoint based on the provided region.
 // https://docs.aws.amazon.com/keyspaces/latest/devguide/programmatic.endpoints.html
 func CassandraEndpointURLForRegion(region string) string {
```
Comment (marcoandredinis) [inline api/utils/aws/endpoint.go:584]: We might need to consider the fips endpoints here as well.
Sorry if this is out of scope

**Review by smallinsky — APPROVED** [review]

File: api/utils/aws/endpoint.go (hunk 2 above)
Comment (smallinsky) [reply api/utils/aws/endpoint.go:584]: This is a helper function that makes a URL based on the region if URI is empty. No fips endpoint is preferred but a  a user can provide fips  endpoint explicitly: `cassandra.us-east-1.amazonaws.com:9142`

File: api/utils/aws/endpoint.go (hunk 1 above)
Comment (greedy52) [reply api/utils/aws/endpoint.go:65]: good catch! will fix

**Review by marcoandredinis — APPROVED** [review]
